diff --git a/admin.lua b/admin.lua
index 61f0141..9757c3e 100644
--- a/admin.lua
+++ b/admin.lua
@@ -15,14 +15,34 @@ $Id: admin.lua 8153 2012-01-06 16:42:02Z jow $
 
 local fs = require "nixio.fs"
 
-m = Map("system", translate("Router Password"),
-	translate("Changes the administrator password for accessing the device"))
+m = Map("system", translate("System Access"),
+        translate("System functions related to system access"))
+local v0 = true -- track password success across maps
+
+-- Allow incorrect root password to prevent settings change
+-- Don't prompt for password if none has been set
+if luci.sys.user.getpasswd("root") then
+        s0 = m:section(TypedSection, "_dummy", translate("Current Password"),
+                translate("Current password required to make changes on this page"))
+        s0.addremove = false
+        s0.anonymous = true
+
+        pw0 = s0:option(Value, "pw0", translate("Current Password"))
+        pw0.password = true
+        -- fail by default
+        v0 = false
+
+        function s0.cfgsections()
+                return { "_pass0" }
+        end
+end
 
-s = m:section(TypedSection, "_dummy", "")
+s = m:section(TypedSection, "_dummy", translate("Router Password"),
+        translate("Changes the administrator password for accessing the device"))
 s.addremove = false
 s.anonymous = true
 
-pw1 = s:option(Value, "pw1", translate("Password"))
+pw1 = s:option(Value, "pw1", translate("New Password"))
 pw1.password = true
 
 pw2 = s:option(Value, "pw2", translate("Confirmation"))
@@ -32,24 +52,36 @@ function s.cfgsections()
 	return { "_pass" }
 end
 
+function m.on_before_commit(map)
+	-- returns true|false
+	if _pass0 then
+		v0 = luci.sys.user.checkpasswd("root", pw0:formvalue("_pass"))
+	end
+
+	if v0 == false then
+		m.message = translate("Incorrect password. Changes rejected!")
+		m.save=v0
+		m2.save=v0
+	end
+end
+
 function m.on_commit(map)
 	local v1 = pw1:formvalue("_pass")
 	local v2 = pw2:formvalue("_pass")
 
-	if v1 and v2 and #v1 > 0 and #v2 > 0 then
-		if v1 == v2 then
-			if luci.sys.user.setpasswd(luci.dispatcher.context.authuser, v1) == 0 then
-				m.message = translate("Password successfully changed!")
-			else
-				m.message = translate("Unknown Error, password not changed!")
-			end
-		else
-			m.message = translate("Given password confirmation did not match, password not changed!")
-		end
-	end
+        if v0 == true and v1 and v2 and #v1 > 0 and #v2 > 0 then
+                if v1 == v2 then
+                        if luci.sys.user.setpasswd(luci.dispatcher.context.authuser, v1) == 0 then
+                                m.message = translate("Password successfully changed!")
+                        else
+                                m.message = translate("Unknown Error, password not changed!")
+                        end
+                else
+                        m.message = translate("Given password confirmation did not match, password not changed!")
+                end
+        end
 end
 
-
 if fs.access("/etc/config/dropbear") then
 
 m2 = Map("dropbear", translate("SSH Access"),
