--- cbi.js	2014-03-17 20:48:16.330196722 +0530
+++ cbi-179.js	2014-03-17 20:48:06.050196901 +0530
@@ -993,6 +993,14 @@
 				if (!(((value.length == 0) && optional) || vstack[0].apply(value, vstack[1])))
 				{
 					// invalid
+					var errorId = field.id; //ID of invalid field
+					var allElements = document.getElementsByTagName('label');	//get all the labels,
+					for (var i = 0; i < allElements.length; i++){				//for all the labels
+					if (allElements[i].getAttribute('for') == errorId){			//get the label whose input is invalid
+						alert(allElements[i].innerHTML + " field is invalid.");	//and create alert with its Value
+						break;
+					    }
+					}
 					field.className += ' cbi-input-invalid';
 					return false;
 				}
